### ISSUE TITLE
Build with flite 2.1.0 fails

### ISSUE THREAD

**Issue by rboy1** [rdp/ffmpeg-windows-build-helpers#604]
USing Ubuntu 18.10 64bit to build 32bit ffmpeg
The current script uses flite 2.1.0
```
build_libflite() {
  download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.1/flite-2.1-release.tar.bz2
  cd flite-2.1-release
    apply_patch file://$patch_dir/flite-2.1.0_mingw-w64-fixes.patch
    if [[ ! -f main/Makefile.bak ]]; then
      sed -i.bak "s/cp -pd/cp -p/" main/Makefile # friendlier cp for OS X
    fi
    generic_configure
    do_make_and_make_install
  cd ..
}
```

This fails when doing a clean build with this error:
> Installing
mkdir -p /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/bin
mkdir -p /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/lib
mkdir -p /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/include/flite
/usr/bin/install -c -m 644 include/*.h /home/c/sandbox/cross_compilers/mingw-w64-i686/i686-w64-mingw32/include/flite
/usr/bin/install: cannot stat '../bin/flite_cmu_us_kal.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_time_awb.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_kal16.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_awb.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_rms.exe': No such file or directory
/usr/bin/install: cannot stat '../bin/flite_cmu_us_slt.exe': No such file or directory
Makefile:136: recipe for target 'install' failed
make[1]: *** [install] Error 1
Makefile:108: recipe for target 'install' failed
make: *** [install] Error 2

However when using an older scripts flite 2.0
```
build_libflite() {
  download_and_unpack_file http://www.festvox.org/flite/packed/flite-2.0/flite-2.0.0-release.tar.bz2
  cd flite-2.0.0-release
    if [[ ! -f configure.bak ]]; then
      sed -i.bak "s|i386-mingw32-|$cross_prefix|" configure
      #sed -i.bak "/define const/i\#include <windows.h>" tools/find_sts_main.c # Needed for x86_64? Untested.
      sed -i.bak "128,134d" main/Makefile # Library only.
      sed -i.bak "s/cp -pd/cp -p/" main/Makefile # friendlier cp for OS X
    fi
    generic_configure
    #cpu_count=1 # can't handle it :|
    do_make_and_make_install
    #cpu_count=$original_cpu_count
  cd ..
}
```

This builds without any issues

**Comment by rdp** [comment]
WSL?

**Comment by rboy1** [comment]
I have no idea what WSL means

**Comment by ericsia** [comment]
> I have no idea what WSL means

it is basically windows 10 linux vm, aka Windows Subsystem for Linux

**Comment by rboy1** [comment]
No. This is a cross compiler build on Linux Ubuntu with a target build for ffmpeg Windows x86 using the mingw cross compiler.

**Comment by rdp** [comment]
That is really weird, seems to work OK ok 20.04 ... you could use docker
(see its folder)

**Comment by rboy1** [comment]
Okay I tried using docker and it failed:
> s2 --enable-libxavs2 --enable-libxavs --extra-cflags=-mtune=generic --extra-cflags=-O3 --enable-static --disable-shared --prefix=/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32 --enable-nonfree --enable-libfdk-aac --enable-decklink ./configure
ERROR: avisynth/avisynth_c.h avisynth/avs/version.h not found
>
>If you think configure made a mistake, make sure you are using the latest
version from Git.  If the latest version fails, report the problem to the
ffmpeg-user@ffmpeg.org mailing list or IRC #ffmpeg on irc.libera.chat.
Include the log file "ffbuild/config.log" produced by configure as this will help
solve the problem.
failed configure ffmpeg_git_with_fdk_aac
[20220214T140821] Build failed. Started: 20220214T140650
[20220214T140821] Stopping container...
ffmpegbuilder
done


BTW, what's the configuration that docker uses? Does it download a prebuilt OS or what's the environment?

**Comment by rdp** [comment]
OK I disabled avisynth for now.  Yeah it does, whacky huh.

**Comment by rboy1** [comment]
Can you share a small write up of the docker configuration maybe in the docker readme folder.. It's not as easy to use docker as it's to build it directly sometimes, especially when there are failures or the scripts to need to customized or new builds required with changes. It doesn't seem to be building with ubuntu 18.10, if it's building with docker and we know what environment docker is using it'll should be easier to replicate the environment and do a direct build

**Comment by Vangelis66** [comment]
>> ERROR: avisynth/avisynth_c.h avisynth/avs/version.h not found

Related issue: #614 :wink:

**Comment by rboy1** [comment]
Well I tried to it again from a fresh Ubuntu 20.04 and it works just fine (the cross compiler script). Don't know what's up with 18.10
It would still be nice to have a little write about how the docker setup.

On an unrelated note @rdp  Roger, is it possible to separate out the install libraries and headers into a separate folder while building ffmpeg? I'm using the libraries to build other projects and want to use the same one used by ffmpeg but it appears to be mixed up with other files in the ffmpeg_git folder.

**Comment by rboy1** [comment]
Is there a way to specify the INSTALL/Libraries/INCLUDE header files folder through a script option? I've looked through the script in the build_ffmpeg() function and can't find any way to define where the INSTALL libraries and header files are specified.

**Comment by rdp** [comment]
I think the "native" build does something like that.  Might be able to
retool the other...maybe... :)

**Comment by SmoochySix4** [comment]
Having the same issue on Arch Linux. Only way for successful build is to comment it out and remove --enable-libflite from the options.
